mod/metrics/set/type/metric/properties.rb

```diff
@@ -45,6 +47,14 @@ def metric_type_property
     field_nest :metric_type, view: :content, items: { view: :name }
   end
 
+  def unit_property
+    field_nest :unit, view: :content, items: { view: :name }
+  end
+
+  def range_property
+    field_nest :range, view: :content, items: { view: :name }
+  end
```

Comment:
this looks like good progress.  of course we need the new javascript to handle showing/hiding.

Reply:
that is not handled in `script_value_type.js.coffee` Lines 8/18?

Reply:
If you've tried it and it works, then I'll have a look.

Normally we don't accept code changes without a test that proves that they work.  It would be good to have a cucumber feature for this.